crates/store/re_chunk_store/src/dataframe.rs

```diff
@@ -398,6 +398,153 @@ impl ComponentColumnDescriptor {
+    //ArchetypeField(ArchetypeFieldColumnSelector),
+}
+
+impl From<ColumnDescriptor> for ColumnSelector {
+    fn from(desc: ColumnDescriptor) -> Self {
+        match desc {
+            ColumnDescriptor::Control(desc) => Self::Control(desc.into()),
+            ColumnDescriptor::Time(desc) => Self::Time(desc.into()),
+            ColumnDescriptor::Component(desc) => Self::Component(desc.into()),
+        }
+    }
+}
+
+impl From<ControlColumnSelector> for ColumnSelector {
+    fn from(desc: ControlColumnSelector) -> Self {
+        Self::Control(desc)
+    }
+}
+
+impl From<TimeColumnSelector> for ColumnSelector {
+    fn from(desc: TimeColumnSelector) -> Self {
+        Self::Time(desc)
+    }
+}
+
+impl From<ComponentColumnSelector> for ColumnSelector {
+    fn from(desc: ComponentColumnSelector) -> Self {
+        Self::Component(desc)
+    }
+}
+
+/// Select a control column.
+///
+/// The only control column currently supported is `rerun.components.RowId`.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct ControlColumnSelector {
+    /// Name of the control column.
+    pub component: ComponentName,
+}
+
+impl From<ControlColumnDescriptor> for ControlColumnSelector {
+    fn from(desc: ControlColumnDescriptor) -> Self {
+        Self {
+            component: desc.component_name,
+        }
+    }
+}
+
+/// Select a time column.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct TimeColumnSelector {
+    /// The name of the timeline.
+    pub timeline: TimelineName,
+}
+
+impl From<TimeColumnDescriptor> for TimeColumnSelector {
+    fn from(desc: TimeColumnDescriptor) -> Self {
+        Self {
+            timeline: desc.timeline.name().clone(),
+        }
+    }
+}
+
+/// Select a component based on its `EntityPath` and `ComponentName`.
+///
+/// Note, that in the future when Rerun supports duplicate tagged components
+/// on the same entity, this selector may be ambiguous. In this case, the
+/// query result will return an Error if it cannot determine a single selected
+/// component.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct ComponentColumnSelector {
+    /// The path of the entity.
+    pub entity_path: EntityPath,
+
+    /// Semantic name associated with this data.
+    pub component: ComponentName,
+
+    /// How to join the data into the `RecordBatch`.
+    pub join_encoding: JoinEncoding,
+}
+
+impl From<ComponentColumnDescriptor> for ComponentColumnSelector {
+    fn from(desc: ComponentColumnDescriptor) -> Self {
+        Self {
+            entity_path: desc.entity_path.clone(),
+            component: desc.component_name,
+            join_encoding: desc.join_encoding,
+        }
+    }
+}
+
+impl ComponentColumnSelector {
+    pub fn new<C: re_types_core::Component>(entity_path: EntityPath) -> Self {
+        Self {
+            entity_path,
+            component: C::name(),
+            join_encoding: JoinEncoding::default(),
+        }
+    }
```

Comment:
Nit: from my perspective, this is a bit too restrictive an API to deserve the privileged `new()` name. Oftentimes, I just have a component name (e.g. when deserialising from blueprint).

```suggestion
    pub fn new(entity_path: EntityPath, component: ComponentName) -> Self {
        Self {
            entity_path,
            component,
            join_encoding: JoinEncoding::default(),
        }
    }

    pub fn new_for_component<C: re_types_core::Component>(entity_path: EntityPath) -> Self {
        Self::new(entity_path, C::name())
    }
```

Reply:
I think you are practically speaking in the minority of users that would feel restricted here.  I think having concrete components is likely to be the more common pattern than dealing with raw ComponentNames.

Reply:
Fair enough.